server/api.go

```diff
@@ -853,3 +855,54 @@ func (p *Plugin) updateSettings(w http.ResponseWriter, r *http.Request, userID s
 	resp, _ := json.Marshal(info.Settings)
 	w.Write(resp)
 }
+
+func (p *Plugin) getIssueByNumber(w http.ResponseWriter, r *http.Request, userID string) {
+
+	ctx := context.Background()
+
+	var githubClient *github.Client
+	owner := r.FormValue("owner")
+	repo := r.FormValue("repo")
+	number := r.FormValue("number")
+	numberInt, _ := strconv.Atoi(number)
+
+	if info, err := p.getGitHubUserInfo(userID); err != nil {
+		writeAPIError(w, err)
+		return
+	} else {
+		githubClient = p.githubConnect(*info.Token)
+	}
+
+	result, _, err := githubClient.Issues.Get(ctx, owner, repo, numberInt)
+	if err != nil {
+		mlog.Error(err.Error())
```

Comment:
Why not `return` here?

Reply:
That's the way it's done on other parts of the code... For instance, [here](https://github.com/mattermost/mattermost-plugin-github/blob/master/server/api.go#L799)

Reply:
That is a good point. While this is consistent, it might cause unexpected behaviour if `result` is `nil`. https://github.com/mattermost/mattermost-plugin-github/pull/252 will address all issue in other parts of the code.

Reply:
So, what do you suggest? Because, currently, if an error occurs `result` would just be `nil`, and then this [line](https://github.com/mattermost/mattermost-plugin-github/pull/258/files/a9fc584c315325868877a4969ba89b624ada6097#diff-0d5a44e9b96f048fa52b477ce6be6253R881) is executed, which ignores any possible error as well.
If I handle that error, there should be an error handler if the marshal occurs... Thoughts?

Reply:
If the code just `return`s here, it's made sure that no invalid response is send. I'm aware that still `200` is returned, but that could be separate ticket.